src/include/txn.i

```diff
@@ -253,52 +232,12 @@ __wt_txn_resolve_prepared_op(
 		 * thing as part of "txn_op2".
 		 */
 
-#ifdef HAVE_DIAGNOSTIC
-		/*
-		 * When an update is not identified for resolution of a
-		 * transaction operation, it might have been already processed
-		 * during the resolution of a previous update belonging to the
-		 * same key. To ascertain transaction tracks multiple extra
-		 * updates processed in resolution of an transaction operation.
-		 */
-		if (upd->prepare_state == WT_PREPARE_RESOLVED) {
-			/*
-			 * FIXME:
-			 * WT_ASSERT(session, txn->multi_update_count > 0);
-			 */
-			--txn->multi_update_count;
-		} else if (upd != op->u.op_upd)
-			++txn->multi_update_count;
-#endif
-
 		if (upd->prepare_state == WT_PREPARE_RESOLVED)
 			break;
-
+		++(*resolved_update_countp);
```

Comment:
You might want to add a `WT_ASSERT` that the new arg is not NULL. Or add a conditional to the increment.